### PR TITLE
[SYCL][L0] check copy operations on interop queue

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: SYCL/Plugin/level_zero_interop_memcpy.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  queue Q2 = sycl::ext::oneapi::level_zero::make<sycl::queue>(
+      Q.get_context(), nativeQ, sycl::ext::oneapi::level_zero::ownership::keep);
+
+  // Command submission works fine
+  Q2.submit([&](handler &cgh) { cgh.single_task<class K>([]() {}); });
+
+  // Copy submissions crashes with discrete GPUs.
+  int hostMem;
+  // int *hostMem = (int*)malloc_host(sizeof(int), Q2);
```
Comment (againull) [inline SYCL/Plugin/level_zero_interop_memcpy.cpp]: I am sorry, I somehow missed this PR.

I am trying to understand what we are trying to check here. Is it expected to fail if hostMem is allocated using malloc_host?

Comment (smaslov-intel) [reply SYCL/Plugin/level_zero_interop_memcpy.cpp]: I think it is only trying to check that "memcpy" at line 27 works OK with interop-queue. I don't recall why "malloc_host" was causing problems. Will try again.

**Comment by againull** [comment]
@smaslov-intel Could you please fix clang-format issue.

File: SYCL/Plugin/level_zero_interop_memcpy.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/backend/level_zero.hpp>
```
Comment (smaslov-intel) [inline SYCL/Plugin/level_zero_interop_memcpy.cpp]:
```suggestion
#include <sycl/ext/oneapi/backend/level_zero.hpp>
```

**Review by againull — APPROVED** [review]

**Comment by smaslov-intel** [comment]
> /verify with [intel/llvm#7253](https://github.com/intel/llvm/pull/7253)

@againull : the test revealed some problem that I am fixing in the compiler

File: SYCL/Plugin/level_zero_interop_memcpy.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <sycl/ext/oneapi/backend/level_zero.hpp>
+#include <level_zero/ze_api.h>
```
Comment (smaslov-intel) [inline SYCL/Plugin/level_zero_interop_memcpy.cpp]:
```suggestion
#include <level_zero/ze_api.h>
#include <sycl/ext/oneapi/backend/level_zero.hpp>
```

**Comment by smaslov-intel** [comment]
The added test passed with https://github.com/intel/llvm/pull/7253
The reported failure cannot be realted:

> [2022-11-02T19:15:56.297Z] Failed Tests (1):
[2022-11-02T19:15:56.297Z]   SYCL :: KernelAndProgram/kernel-bundle-merge-options-env.cpp

Please consider merging this

**Pull request merged**